hastory-server-data/src/Hastory/Server/Digest.hs

```diff
@@ -0,0 +1,21 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Hastory.Server.Digest where
+
+import Crypto.Hash (Digest, HashAlgorithm, digestFromByteString)
+import qualified Data.ByteArray as BA
+import qualified Data.ByteString as B
+import Data.Proxy (Proxy(Proxy))
+import Database.Persist.Class (PersistField(fromPersistValue, toPersistValue))
+import Database.Persist.Sql (PersistFieldSql(sqlType))
+import Database.Persist.Types (PersistValue(PersistByteString))
+
+instance HashAlgorithm a => PersistField (Digest a) where
+  toPersistValue = toPersistValue . B.pack . BA.unpack
+  fromPersistValue (PersistByteString rawDigest) =
+    maybe (Left "Unable to reify Digest from ByteString") Right (digestFromByteString rawDigest)
```

Comment:
Nit: I would just write this out using `case .. of` instead of using `maybe`, but it doesn't matter. This is fine.

Reply:
Removed use of `maybe`.